tools/version.sh

```diff
@@ -86,7 +86,11 @@ done
 OUTFILE=$1
 
 if [ -z ${VERSION} ] ; then
-  VERSION=`git -C ${WD} tag --sort=taggerdate | tail -1 | cut -d'-' -f2`
+  VERSION=`git -C ${WD} describe --exact-match 2>/dev/null | tail -1 | cut -d'-' -f2`
```

Comment:
should we remove --exact-match? otherwise git will complain if the new patch add to the branch after tagging.

Reply:
This issue is if we remove --exact-match, git describe won't fail anymore if there is no tag on the branch.  So in master we will always have 8.2

Reply:
should we check 8.2 and fallback to the sorting too?

Reply:
That was my recommendation.  Alternatively if it is easier checking for nuttx-#.#.#

Reply:
I didn't want to do tests for had coded versions, but it looks like we have to.  Please take a look.

Also, I removed the part that appends a patch to the older < 8.2.  I don't think we need that anymore.

Reply:
LGTM.